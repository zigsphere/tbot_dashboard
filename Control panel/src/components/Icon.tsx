import * as React from "react";

export const Icon = () => <div></div>;
