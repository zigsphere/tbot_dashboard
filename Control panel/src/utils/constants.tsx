import { MenuGroup, MenuItem } from "./types";

export const menuItems = (guildId: string): MenuGroup[] => [
  {
    name: "Security",
    path: "/security",
    routes: [
      {
        name: "Authentication",
        path: `/dashboard/${guildId}/security/authentication`,
      },
      {
        name: "Roles",
        path: `/dashboard/${guildId}/security/roles`,
      },
    ],
  },
  {
    name: "General",
    path: "/general",
    routes: [
      {
        name: "General",
        path: `/dashboard/${guildId}/general/muted`,
      },
    ],
  },
  {
    name: "Miscellaneous",
    path: "/miscellaneous",
    routes: [
      {
        name: "Commands",
        path: `/dashboard/${guildId}/miscellaneous/fun-commands`,
      },
      {
        name: "Levels",
        path: `/dashboard/${guildId}/miscellaneous/levels`,
      },
    ],
  },
];

export const selectMenuItems: MenuItem[] = [
  { name: "Developer", value: "developer" },
  { name: "Moderator", value: "moderator" },
  { name: "Guest", value: "guest" },
  { name: "React", value: "react" },
  { name: "Angular", value: "angular" },
  { name: "Node", value: "node" },
  { name: "Java", value: "java" },
];
