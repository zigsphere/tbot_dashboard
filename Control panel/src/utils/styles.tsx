import styled, { css } from "styled-components";

export const SidebarStyle = styled.aside`
  width: 100px;
  background-color: ${({ theme }) => theme.base.sidebar};
  color: ${({ theme }) => theme.text.main};
  height: 100%;
  font-family: "Poppins";
  box-sizing: border-box;
  position: fixed;
  overflow: scroll;
  ::-webkit-scrollbar {
    display: none;
  }
`;

export const MenuStyle = styled.aside`
  width: 350px;
  background-color: ${({ theme }) => theme.base.menu};
  color: ${({ theme }) => theme.text.main};
  height: 100%;
  font-family: "Poppins";
  box-sizing: border-box;
  position: fixed;
  margin-left: 100px;
  overflow: scroll;
  ::-webkit-scrollbar {
    display: none;
  }
`;

export const BasePageStyle = styled.div`
  min-height: 100%;
  background-color: ${({ theme }) => theme.base.content};
  color: ${({ theme }) => theme.text.main};
  font-family: "Poppins";
  box-sizing: border-box;
  position: absolute;
  margin-left: 450px;
  width: calc(100% - 450px);
`;

export const DashboardPageStyle = styled.div``;

export const IconStyle = styled.div`
  height: 60px;
  width: 60px;
  background-color: #c4c4c4;
  border-radius: 50px;
`;

// Headers
export const BaseHeader = styled.header`
  box-sizing: border-box;
  height: ${({ theme }) => theme.height.topContent};
  box-shadow: 0px 1px 25px 0px rgba(0, 0, 0, 0.25);
  display: flex;
  align-items: center;
`;
export const SidebarHeader = styled(BaseHeader)`
  justify-content: center;
`;
export const DashboardHeader = styled(BaseHeader)`
  padding: 25px;
`;
export const MenuHeader = styled(BaseHeader)`
  padding: 25px;
  position: fixed;
  z-index: 9999;
  background-color: inherit;
  width: 100%;
  :hover {
    cursor: pointer;
  }
`;

export const SidebarContent = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: ${({ theme }) => theme.padding.content};
`;

export const GuildIcon = styled(IconStyle)<any>`
  height: 75px;
  width: 75px;
  margin: 5px 0px;
  background: url(${({ bg }) => bg});
  background-color: grey;
  background-position: center;
  background-size: 100%;
`;

export const MenuContent = styled.div`
  box-sizing: border-box;
  padding: ${({ theme }) => theme.padding.content};
  margin-top: 100px;
`;

export const MenuCategory = styled.div`
  box-sizing: border-box;
  font-size: 16px;
  font-weight: bold;
  :nth-child(2) {
    margin: 20px 0px;
  }
  span {
    padding: 0px 20px;
  }
`;
export const MenuCategoryItem = styled.section`
  font-size: 22px;
  font-weight: lighter;
  padding: 5px 0px;
  transition: 100ms background-color;
  :hover {
    cursor: pointer;
    background-color: #42494c;
  }
  span {
    padding: 0px 20px;
  }
`;

export const DropdownHeader = styled.div`
  background-color: #21262d;
  padding: 8px 20px;
  border-radius: 5px;
  font-size: 16px;
  font-weight: bold;
  border-bottom: 0.5px solid rgba(255, 255, 255, 0.1);
  display: flex;
  justify-content: space-between;
  cursor: pointer;
`;

export const DropdownItemContainer = styled.div<any>`
  max-height: 0px;
  overflow-y: scroll;
  opacity: 0;
  transition: 300ms;
  ::-webkit-scrollbar {
    width: 5px;
  }
  ::-webkit-scrollbar-track {
  }
  ::-webkit-scrollbar-thumb {
    background-color: #171717;
    border-radius: 50px;
  }
  ${(props) =>
    props.expanded
      ? css`
          max-height: 220px;
          opacity: 1;
        `
      : css`
          max-height: 0px;

          opacity: 0;
        `}
`;

export const DropdownItem = styled.div<any>`
  background-color: #21262d;
  color: white;
  border-bottom: 0.5px solid rgba(255, 255, 255, 0.1);
  padding: 5px 20px;
  :hover {
    background-color: #545d6b;
    cursor: pointer;
  }
  ${(props) =>
    props.isSelected &&
    css`
      background-color: #2e2e2e;
    `}
`;

export const DropdownFooter = styled.footer`
  background-color: #21262d;
  padding: 8px 20px;
  border-bottom-left-radius: 5px;
  border-bottom-right-radius: 5px;
  max-height: 42px;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const Button = styled.button`
  font-family: "Poppins";
  padding: 5px 12px;
  background-color: inherit;
  border: 1px solid rgba(255, 255, 255, 0.15);
  outline: none;
  color: white;
  border-radius: 5px;
  transition: 100ms;
  :hover {
    cursor: pointer;
    background-color: #4a4a4a;
  }
  :active {
    background-color: #262626;
  }
  :disabled {
    background-color: #919191;
    border: none;
  }
`;

export const Input = styled.input`
  font-family: "Poppins";
  outline: none;
  border: 1px solid rgba(255, 255, 255, 0.15);
  padding: 4px 8px;
  border-radius: 5px;
  background-color: inherit;
  color: white;
`;
