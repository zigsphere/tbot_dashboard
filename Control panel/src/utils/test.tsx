import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { DarkTheme } from "./themes";

const Providers = ({ children }: any) => {
  return <ThemeProvider theme={DarkTheme}>{children}</ThemeProvider>;
};

const customRender = (ui: any, options?: any) =>
  render(ui, { wrapper: Providers, ...options });

export * from "@testing-library/react";
export { customRender as render };
