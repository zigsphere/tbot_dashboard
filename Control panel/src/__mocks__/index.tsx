import { Guild } from "../utils/types";

export const guilds: Guild[] = [
  {
    id: "123",
    name: "Developers",
    icon: "",
  },
  {
    id: "124",
    name: "Developers",
    icon: "",
  },
  {
    id: "125",
    name: "Developers",
    icon: "",
  },
  {
    id: "126",
    name: "Developers",
    icon: "",
  },
  {
    id: "127",
    name: "Developers",
    icon: "",
  },
  {
    id: "128",
    name: "Developers",
    icon: "",
  },
  {
    id: "129",
    name: "Developers",
    icon: "",
  },
  {
    id: "130",
    name: "Developers",
    icon: "",
  },
  {
    id: "131",
    name: "Developers",
    icon: "",
  },
  {
    id: "132",
    name: "Developers",
    icon: "",
  },
  {
    id: "133",
    name: "Developers",
    icon: "",
  },
  {
    id: "134",
    name: "Developers",
    icon: "",
  },
  {
    id: "135",
    name: "Developers",
    icon: "",
  },
  {
    id: "136",
    name: "Developers",
    icon: "",
  },
  {
    id: "137",
    name: "Developers",
    icon: "",
  },
  {
    id: "138",
    name: "Developers",
    icon: "",
  },
];
