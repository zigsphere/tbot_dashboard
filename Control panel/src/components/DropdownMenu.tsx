import * as React from "react";
import { createRef, useState } from "react";
import {
  Button,
  DropdownFooter,
  DropdownHeader,
  DropdownItem,
  DropdownItemContainer,
  Input,
} from "../utils/styles";
import { MenuItem } from "../utils/types";

type Props = {
  title: string;
  items: MenuItem[];
};

export const DropdownMenu = ({ title, items }: Props) => {
  const menuRef = createRef<HTMLDivElement>();
  const [selected, setSelected] = useState<MenuItem | undefined>();
  const [expanded, setExpanded] = useState<boolean>(false);
  const [search, setSearch] = useState<string>("");
  const [searchResults, setSearchResults] = useState<MenuItem[]>(items);

  const handleChange = (event: React.FormEvent<HTMLInputElement>) => {
    const { value } = event.currentTarget;
    setSearch(value);
    const filtered = items.filter((item) =>
      item.name.toLowerCase().includes(value.toLowerCase())
    );
    setSearchResults(filtered);
  };

  const resetState = () => {
    setExpanded(false);
    setSearchResults(items);
    setSelected(undefined);
    setSearch("");
  };

  return (
    <div>
      <DropdownHeader onClick={() => setExpanded(!expanded)}>
        <p>{title}</p>
        <p>{selected && selected.name}</p>
      </DropdownHeader>
      <DropdownItemContainer ref={menuRef} expanded={expanded}>
        {searchResults.map((item) => (
          <DropdownItem
            key={item.value}
            onClick={() => setSelected(item)}
            isSelected={selected && item.value === selected.value}
          >
            {item.name}
          </DropdownItem>
        ))}
      </DropdownItemContainer>
      {expanded && (
        <DropdownFooter>
          <div>
            <Button style={{ marginRight: "5px" }} disabled={!selected}>
              Save
            </Button>
            <Button onClick={resetState}>Collapse</Button>
          </div>
          <div>
            <Input
              onChange={handleChange}
              value={search}
              placeholder="Search"
            />
          </div>
        </DropdownFooter>
      )}
    </div>
  );
};
