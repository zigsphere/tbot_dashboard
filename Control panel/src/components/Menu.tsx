import * as React from "react";
import { useContext } from "react";
import {
  MenuStyle,
  MenuHeader,
  MenuContent,
  MenuCategory,
  MenuCategoryItem,
} from "../utils/styles";
import { menuItems } from "../utils/constants";
import { History } from "history";
import GuildContext from "../utils/contexts/GuildContext";

type MenuProps = {
  title: string;
  history: History;
};

export const Menu = ({ title, history }: MenuProps) => {
  const { guild } = useContext(GuildContext);
  if (!guild)
    return (
      <MenuStyle>
        <MenuHeader>Please Select a Guild</MenuHeader>
      </MenuStyle>
    );

  return (
    <MenuStyle>
      <MenuHeader onClick={() => history.push(`/dashboard/${guild.id}`)}>
        {title}
      </MenuHeader>
      <MenuContent>
        {menuItems(guild.id).map((item) => (
          <MenuCategory key={item.name}>
            <span>{item.name}</span>
            {item.routes.map((route) => (
              <MenuCategoryItem
                key={route.name}
                onClick={() => history.push(route.path)}
              >
                <span># {route.name}</span>
              </MenuCategoryItem>
            ))}
          </MenuCategory>
        ))}
      </MenuContent>
    </MenuStyle>
  );
};
