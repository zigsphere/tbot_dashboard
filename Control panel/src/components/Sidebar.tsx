import * as React from "react";
import { useContext } from "react";
import GuildContext from "../utils/contexts/GuildContext";
import {
  SidebarStyle,
  SidebarHeader,
  IconStyle,
  SidebarContent,
  GuildIcon,
} from "../utils/styles";
import { Guild } from "../utils/types";
import { History } from "history";

type SidebarProps = {
  guilds: Guild[];
  history: History;
};

export const Sidebar = (props: SidebarProps) => {
  const { setGuild } = useContext(GuildContext);
  return (
    <SidebarStyle>
      <SidebarHeader>
        <IconStyle />
      </SidebarHeader>
      <SidebarContent>
        {props.guilds.map((guild) => (
          <GuildIcon
            bg={guild.icon}
            className="guild-icon"
            key={guild.id}
            onClick={() => {
              setGuild(guild);
              console.log(guild.icon);
              props.history.push(`/dashboard/${guild.id}`);
            }}
          />
        ))}
      </SidebarContent>
    </SidebarStyle>
  );
};
