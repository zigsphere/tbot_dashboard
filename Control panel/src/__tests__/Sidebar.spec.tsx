/* eslint-disable jest/no-mocks-import */
import * as React from "react";
import { render } from "../utils/test";
import { Sidebar } from "../components/Sidebar";
import { guilds } from "../__mocks__";
import { useHistory } from "react-router-dom";
import { History } from "history";

let history: History;

jest.mock("react-router-dom", () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

beforeEach(() => {
  history = useHistory();
});

it("should render Sidebar", () => {
  const { asFragment } = render(<Sidebar guilds={guilds} history={history} />);
  const firstRender = asFragment();
  expect(firstRender).toMatchSnapshot();
});

it("should render sidebar with 16 elements", () => {
  const { container } = render(<Sidebar history={history} guilds={guilds} />);
  const div = container.getElementsByClassName("guild-icon");
  expect(div.length).toBe(16);
});
