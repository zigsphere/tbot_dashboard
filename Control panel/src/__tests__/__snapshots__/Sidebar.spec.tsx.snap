// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Sidebar 1`] = `
<DocumentFragment>
  <aside
    class="sc-bdfBwQ liFhOG"
  >
    <header
      class="sc-jSgupP sc-gKsewC hFNlhE hXkiSa"
    >
      <div
        class="sc-eCssSg faLJyX"
      />
    </header>
    <div
      class="sc-pFZIQ jppQcH"
    >
      <div
        class="sc-eCssSg sc-jrAGrp faLJyX jXUtwz guild-icon"
      />
      <div
        class="sc-eCssSg sc-jrAGrp faLJyX jXUtwz guild-icon"
      />
      <div
        class="sc-eCssSg sc-jrAGrp faLJyX jXUtwz guild-icon"
      />
      <div
        class="sc-eCssSg sc-jrAGrp faLJyX jXUtwz guild-icon"
      />
      <div
        class="sc-eCssSg sc-jrAGrp faLJyX jXUtwz guild-icon"
      />
      <div
        class="sc-eCssSg sc-jrAGrp faLJyX jXUtwz guild-icon"
      />
      <div
        class="sc-eCssSg sc-jrAGrp faLJyX jXUtwz guild-icon"
      />
      <div
        class="sc-eCssSg sc-jrAGrp faLJyX jXUtwz guild-icon"
      />
      <div
        class="sc-eCssSg sc-jrAGrp faLJyX jXUtwz guild-icon"
      />
      <div
        class="sc-eCssSg sc-jrAGrp faLJyX jXUtwz guild-icon"
      />
      <div
        class="sc-eCssSg sc-jrAGrp faLJyX jXUtwz guild-icon"
      />
      <div
        class="sc-eCssSg sc-jrAGrp faLJyX jXUtwz guild-icon"
      />
      <div
        class="sc-eCssSg sc-jrAGrp faLJyX jXUtwz guild-icon"
      />
      <div
        class="sc-eCssSg sc-jrAGrp faLJyX jXUtwz guild-icon"
      />
      <div
        class="sc-eCssSg sc-jrAGrp faLJyX jXUtwz guild-icon"
      />
      <div
        class="sc-eCssSg sc-jrAGrp faLJyX jXUtwz guild-icon"
      />
    </div>
  </aside>
</DocumentFragment>
`;
